docs/rfcs/001-updateable-bundled-packages.md

```diff
@@ -0,0 +1,100 @@
+
+If a package wants to use API features of a newer version of Atom while still supporting older Atom versions, it must do so in a way that is aware of the user's version and adjust itself accordingly.
+
+## Drawbacks
+
+> Why should we *not* do this?
+
+The primary drawback of this approach is that updateable bundled packages might exhibit problems on older Atom versions due to missing or changed APIs in Atom, Electron, or Node.js.  The solution for these packages is to keep their `engines` field updated appropriately, but there's still a chance that some updates will slip through without the necessary engine version changes.
+
+One other possible drawback is that an updated version of a bundled package might not be compatible across two different Atom channels.  For example, if the user installs a new update to a bundled package that only supports the current Atom Beta release or higher, the user will no longer have access to that package if they open Atom Stable.
+
+However, this drawback is no different than what the user would face today installing a community package under the same circumstances, so this could be considered a general problem in the Atom package ecosystem.
+
+## Rationale and alternatives
+
+> - Why is this approach the best in the space of possible approaches?
+> - What other approaches have been considered and what is the rationale for not choosing them?
+> - What is the impact of not doing this?
+
+This is the best approach for updating bundled packages because it allows those packages to take control of their own release cycle so long as they manage their Atom engine version correctly.  It also does so in a way that allows us to decide which packages can be updated independently, reducing the likelihood of problems for users.
+
+The primary alternative to this approach is to speed up the Atom release cycle so that bundled Atom package updates will reach users more frequently.  This approach will be investigated independently of this RFC as it may still be valuable even with updateable bundled packages.
+
+## Unresolved questions
+
+> - What unresolved questions do you expect to resolve through the RFC process before this gets merged?
+
+Is it enough to just depend on the `engines` field of `package.json` to protect users from installing a package update that doesn't work with their version of Atom?
+
+Is `updateable` the right name for the field in `package.json`?  Is there a clearer name?
+
+> - What unresolved questions do you expect to resolve through the implementation of this feature before it is released in a new version of Atom?
+
+Can package authors ship updates to stable-only and beta-only versions of their packages simultaneously?  For example, can the GitHub package keep shipping hotfixes to 0.14.x which targets Atom >=1.27.0 while also shipping updates to 0.15.x which targets >=1.28.0?
```

Comment:
> Can package authors ship updates to stable-only and beta-only versions of their packages simultaneously?

I don't think anything prevents us from _shipping_ versions in this way. The issue is more about how it affects people installing and using these versions, right? For example, if I update the github package, that update is presumably stored at `~/.atom/packages/github`. If the version at `~/.atom/packages/github` is only compatible with Atom >=1.28, and I open Atom 1.27.0, what happens?

Reply:
Yep, I phrased that from the wrong perspective, will update it accordingly.

Reply:
Actually, based on someting @smashwilson [pointed out before](https://github.com/atom/atom/pull/17379#discussion_r189599200) I think the scenario I'm positing here isn't even possible so I'll just remove it :scissors: